modules/org_policy_v2/main.tf

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Locals configuration for module logic
+ *****************************************/
+locals {
+  org_policies = {
+    for k, v in var.org_policies : k => merge(v, { rules = [for k1, v1 in v.rules :
+      merge(v1, { allow_list_length = length(v1.allow), deny_list_length = length(v1.deny), enforcement = length(v1.allow) > 0 || length(v1.deny) > 0 ? null : v1.enforcement, values = [{ allow = v1.allow, deny = v1.deny }] })
+    ] })
+  }
+}
+
+/******************************************
+  Invoke constraints module
+ *****************************************/
+module "gcp_org_policy" {
```

Comment:
It seems like this is just a thin wrapper around the other module. To simplify, can we start with just supporting 1 org policy per instantiation? In the example we can show how to use for_each to manage multiple instantiations of this module. This will help simplify the complex object required as an input and would be more aligned with the top level module https://github.com/terraform-google-modules/terraform-google-org-policy/tree/master#usage

Reply:
Okay, which will lead to slight change in the logic. I thought of future use case to implement **CIS Org Policies**, which will use this module. I will rethink on this.